OpenRA.Game/Traits/Util.cs

```diff
@@ -146,5 +146,153 @@ public static int ApplyPercentageModifiers(int number, IEnumerable<int> percenta
 
 			return (int)a;
 		}
+
+		// Algorithm obtained from ftp://ftp.isc.org/pub/usenet/comp.sources.unix/volume26/line3d
+		public static IEnumerable<CPos> Raycast(CPos source, CPos target)
```

Comment:
I'll need help with this. Doing it for the rectangular grid would be easy enough, but for the iso/rhombus grid makes the problem more difficult for me to think about.